pkg/test/utils/kubernetes.go

```diff
@@ -679,8 +679,10 @@ func NewPod(name, namespace string) runtime.Object {
 // NewCRDv1 creates a new CRD object of version v1
 func NewCRDv1(t *testing.T, name, group, resourceKind string, resourceVersions []string) runtime.Object {
 	crdVersions := []interface{}{}
+	served := true
 	for _, v := range resourceVersions {
-		crdVersions = append(crdVersions, interface{}(map[string]interface{}{"name": interface{}(v)}))
+		crdVersions = append(crdVersions, interface{}(map[string]interface{}{"name": interface{}(v), "served": interface{}(served)}))
+		served = false
```

Comment:
I don't like the crd handling to be done like this but this code will ultimately go away in the next PR (see description)